src/geom.cpp

```diff
@@ -57,7 +57,7 @@ void simplify(GeometryType const &input, GeometryType &output, double max_distan
 				min_distance = std::min(min_distance, boost::geometry::distance(line, i));
 
             std::size_t query_expected = ((start == 0 || end == input.size() - 1) ? 2 : 4);
-            if(result.size() == query_expected && min_distance > max_distance) {
+            if(result.size() == query_expected && min_distance < max_distance) {
```

Comment:
This check was added to ensure that the new line does not get too close to other (non-related) lines of the polygon. Due to rounding of the coordinates, this may result in a self intersection. The check was correct, but maybe max_distance is too strict.

Reply:
For example on these fjords and lochs, you have two sides which are very close to each other. After simplification, the lines of these polygons may be even closer. Not intersecting, but maybe after rounding of the coordinates, they do intersect.

Reply:
Ah. If I add this:

> if (min_distance!=0) { std::cout << "min_distance is " << min_distance << std::endl; }

then there's no output... i.e. min_distance is always 0.

Reply:
Yes, i think i know why this is. A query is performed to obtain the 5 nearest lines of the polygon:

Reply:
But that would suggest that indeed the simplifier is not simplifying at all.

Reply:
Yes - I think that's the case. Current master at z10 looks like the original geometry:

![Screenshot 2021-06-07 at 12 57 56](https://user-images.githubusercontent.com/694425/121013170-6940bb00-c790-11eb-94e0-5697950212ab.png)

but with the check removed (i.e. equivalent to this PR):

![Screenshot 2021-06-07 at 13 00 21](https://user-images.githubusercontent.com/694425/121013232-7bbaf480-c790-11eb-90b2-330370fd9bcc.png)

which is closer to what I'd expect.

Reply:
You can try removing the check, but this may cause issues again on these islands of scotland. That is something to check.

Reply:
I think you can actually remove this whole check, so these lines: 

https://github.com/systemed/tilemaker/blob/fix_simplify/src/geom.cpp#L50-L57

And change:
https://github.com/kleunen/tilemaker/blob/master/src/geom.cpp#L60

from: 
````C++
-if(result.size() == query_expected && min_distance > max_distance) {
````

to:
````C++
+if(result.size() == query_expected) {
````

I checked on this scotland export, it seems to work fine even without this distance check. North Uist is not displayed at zoom level 4, but is displayed at other zoom levels. I think this is how we eventually accepted the PR.